pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -111,17 +126,33 @@ func (a Adapter) Push(path string, ignoredFiles []string, forceBuild bool, globE
 		isForcePush = true
 	}
 
+	podSelector := fmt.Sprintf("component=%s", a.ComponentName)
+	watchOptions := metav1.ListOptions{
+		LabelSelector: podSelector,
+	}
+	// Wait for Pod to be in running state otherwise we can't sync data or exec commands to it.
+	pod, err := a.Client.WaitAndGetPod(watchOptions, corev1.PodRunning, "Waiting for component to start")
+	if err != nil {
+		return errors.Wrapf(err, "error while waiting for pod  %s", podSelector)
+	}
+	a.pod = pod
```

Comment:
Since `WaitAndGetPod` doesn't wait for a new pod to rollout and for the old one to go away, there could be a race-condition here where the old pod is returned (before it was moved to the terminating state) instead of the new one.

This may be more of a matter of waiting for https://github.com/openshift/odo/pull/2731 to get merged and to use the "Wait for Rollout" function instead.

Reply:
#2731 adds the wait for rollout to this exact function so we shouldn't need to add any additional waits on this function.